# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ImageEncoder.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 dom/base/ImageEncoder.h
--- a/dom/base/ImageEncoder.h	Fri Jun 12 21:53:14 2020 +0000
+++ b/dom/base/ImageEncoder.h	Mon Jun 15 18:25:55 2020 +0000
@@ -20,7 +20,7 @@
 namespace mozilla {
 
 namespace layers {
-class AsyncCanvasRenderer;
+class CanvasRenderer;
 class Image;
 }  // namespace layers
 
@@ -40,7 +40,7 @@
   static nsresult ExtractData(nsAString& aType, const nsAString& aOptions,
                               const nsIntSize aSize, bool aUsePlaceholder,
                               nsICanvasRenderingContextInternal* aContext,
-                              layers::AsyncCanvasRenderer* aRenderer,
+                              layers::CanvasRenderer* aRenderer,
                               nsIInputStream** aStream);
 
   // Extracts data asynchronously. aType may change to "image/png" if we had to
@@ -86,7 +86,7 @@
       const nsAString& aType, const nsAString& aOptions, uint8_t* aImageBuffer,
       int32_t aFormat, const nsIntSize aSize, bool aUsePlaceholder,
       layers::Image* aImage, nsICanvasRenderingContextInternal* aContext,
-      layers::AsyncCanvasRenderer* aRenderer, nsIInputStream** aStream,
+      layers::CanvasRenderer* aRenderer, nsIInputStream** aStream,
       imgIEncoder* aEncoder);
 
   // Creates and returns an encoder instance of the type specified in aType.